doped/utils/parsing.py

```
@@ -1988,3 +2009,374 @@ def _multiple_files_warning(file_type, directory, chosen_filepath, action=None,
+                    vasprun_obj.vbm = band_edge_prop.vbm_info.as_dict()["energy"]
+                    vasprun_obj.cbm = band_edge_prop.cbm_info.as_dict()["energy"]
+                    vasprun_obj.band_gap = vasprun_obj.cbm - vasprun_obj.vbm
+        else:
+            raise ValueError("Use doped or pymatgen for finding band_gap")
+        return vasprun_obj
+
+
+    @classmethod
+    def _get_bulk_locpot_dict(cls, bulk_path, quiet=False):
+
+        bulk_locpot_path, multiple = _get_output_files_warn_if_multiple(filename, bulk_path, dir_type="bulk")
```

Ruff (0.12.2)

Undefined name `filename`

(F821)